### PR TITLE
refactor: check document and window object to support nextjs rendering

### PR DESCRIPTION
**Pull request by prathapsimha**
Contributes to #

changes check document and window object to support nextjs rendering

#### What did you change?
modified:  packages/ibm-products/src/components/Datagrid/Datagrid/DatagridSelectAll.js
modified:   packages/ibm-products/src/components/Datagrid/useInfiniteScroll.js
modified:   packages/ibm-products/src/components/Datagrid/useSelectRows.js

#### How did you test and verify your work?
running fine in local when I integrate with my nextjs project. please verify

### CODE REVIEW

**Comment by prathapsimha** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by prathapsimha** [comment]
recheck

**Comment by prathapsimha** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/ibm-products/src/components/Datagrid/useInfiniteScroll.js [hunk 1]
```diff
@@ -23,7 +23,10 @@ const useInfiniteScroll = (hooks) => {
       tableId,
       loadMoreThreshold,
     } = instance;
-    const tableElement = document.querySelector(`#${tableId}`);
+    var tableElement;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/Datagrid/useInfiniteScroll.js:26]: Can you change this to `let`?

Comment (prathapsimha) [reply packages/ibm-products/src/components/Datagrid/useInfiniteScroll.js]: Changed var to let

**Comment by matthewgallo** [comment]
One more thing, the DCO check requires the email associated with your commits to be added to your GitHub account/profile. Can you double check that?

> prathap simha seems not to be a GitHub user. You need a GitHub account to be able to sign the DCO. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by prathapsimha** [comment]
I have added email, but still DCO is failing

**Comment by prathapsimha** [comment]
recheck

**Comment by elycheea** [comment]
I believe this is the issue. The commit history shows two different IDs. We’ve seen this issue before, I think the easiest way around is probably to create a new PR with your changes. These commits aren’t tied to your email address at all (you can check the email with `git show <commit-hash>`). You can either amend the commits (which can sometimes feel dirtier) or create a new PR with the changes.

> image

> image

Since this is on your main branch already, I would recommend creating a new branch from our `main` branch (`upstream/main`) and resubmitting.

**Pull request closed without merging**